### DISCUSSION TITLE
Access fastapi session cookie within nicegui sub page?

### DISCUSSION THREAD

**Discussion by philippgl** [zauberzeug/nicegui#3054] [Q&A]
### Question

I want to create an admin sub page for my website. Can I use the fastapi/starlette session cookie for authentication?

```python
from fastapi import FastAPI
from nicegui import app, ui
from starlette.requests import Request

def init_admin(fastapi_app: FastAPI) -> None:
    @ui.page("/")
    def show(request: Request) -> None:
        ui.label("Hello, FastAPI!")

        ui.dark_mode().bind_value(app.storage.user, "dark_mode")
        ui.checkbox("dark mode").bind_value(app.storage.user, "dark_mode")

    ui.run_with(
        fastapi_app,
        mount_path="/admin",  # NOTE this can be omitted if you want the paths passed
        storage_secret="foobarbaz",
    )
```

Now the `request` object refers to the nicegui request. Is there a way to access to the session cookie of the outer fastapi app? Read-only would be good enough in my case. I can see the cookie in the upgrade request, but have not found a way to access it.

**Comment by hasansezertasan** [comment]
> I want to create an admin sub page for my website. Can I use the fastapi/starlette session cookie for authentication?

If you go deep, of course, you can 🦾. The ❓ question is how deep you should go.

You can implement your own middleware replicating the existing authentication flow and mount it to your sub app.

> Now the `request` object refers to the nicegui request. Is there a way to access to the session cookie of the outer fastapi app? Read-only would be good enough in my case. I can see the cookie in the upgrade request, but have not found a way to access it.

Doesn't NiceGUI's Request object inherit Starlette/FastAPI Request object? If yes, you should be able to access `request.session` or `request.state.xyz`.
